fix samples endpoint

ttung opened the pull request:
### Description
This allows us to not execute N queries during the samples endpoint for the gisaid data.

Depends on #361 

### Test plan
turned on profiling and verified that the samples endpoint executes 3 queries now.

shanaxel42 reviewed (approved):
ooof I knew I slowed this down!!

Commit:
fix samples endpoint

### Description
This allows us to not execute N queries during the samples endpoint for the gisaid data.

### Test plan
turned on profiling and verified that the samples endpoint executes 3 queries now.
